### PR TITLE
Add result page.

### PR DESCRIPTION
**Pull request by kyudori**
## Description

LPVS Front-End `Reseult` Page.
Result Page provides `PR informatin` and `detected file list`.

Dashboard(last page) pages will be uploaded.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup/refactoring
- [ ] Documentation update
- [x] This change requires a documentation update
- [ ] CI system update
- [ ] Test Coverage update

# How Has This Been Tested?
1. Update environment variables of `docker-compose.yml` file.
2. Build the lpvs container : `docker compose build`
3. Run containers: `docker compose up`
4. See the lpvs home page with any browser: `http://localhost:7896`

For using social login with Naver, Kakao, and Google, fill the necessary information:
- Update the `application.properties` located in `src/main/resources`:
Enter client-id and client-secret. (Kakao doesn't require a client-secret.)
- Update the `.env` file located in `frontend`:
Enter your REST_API_KEY.

The values required to implement the login function can be obtained from the link below.
Naver: https://developers.naver.com/products/login/api/api.md
Kakao: https://developers.kakao.com/docs/latest/ko/kakaologin/common
Google: https://cloud.google.com/identity-platform/docs/web/google

result

image

**Test Configuration**:
* Java: v11
* LPVS Release: v1.1.0

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/pages/History.jsx [hunk 1]
```diff
@@ -188,7 +188,7 @@ export const History= () => {
             </div>
 
             <div className="my-option-header">
-              <img className="my-option-line" alt="My option line" src="/image/svg/historyMyOptionLine.svg" />
+              <img className="my-option-line" alt="My option line" src="/image/svg/HistoryMyOptionLine.svg" />
```
Comment (tiokim) [inline frontend/src/pages/History.jsx:191]: `historyMyOptionLine.svg` should be renamed according to this change.

File: frontend/src/pages/Result.jsx [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+                <div className={`text-wrapper-26 ${lpvsResult?.lpvsResultFileList[3]?.path.length >= 28 ? 'long' : 'short'}`}>{lpvsResult?.lpvsResultFileList[3]?.path}</div>
+              </div>
+              <div className="text-wrapper-27-container">
+                <div className={`text-wrapper-27 ${lpvsResult?.lpvsResultFileList[2]?.path.length >= 28 ? 'long' : 'short'}`}>{lpvsResult?.lpvsResultFileList[2]?.path}</div>
+              </div>
+              <div className="text-wrapper-28-container">
+                <div className={`text-wrapper-28 ${lpvsResult?.lpvsResultFileList[1]?.path.length >= 28 ? 'long' : 'short'}`}>{lpvsResult?.lpvsResultFileList[1]?.path}</div>
+              </div>
+              <div className="text-wrapper-29-container">
+                <div className={`text-wrapper-29 ${lpvsResult?.lpvsResultFileList[0]?.path.length >= 28 ? 'long' : 'short'}`}>{lpvsResult?.lpvsResultFileList[0]?.path}</div>
```
Comment (tiokim) [inline frontend/src/pages/Result.jsx:306]: If the `path` value is `NULL`, it goes to the error.

**TEST CASE** : https://github.com/tiokim/edge-home-orchestration-go/pull/100
**DB Records**
```
mysql> select * from detected_license where pull_request_id=11;
+----+-----------------+------------+-------------+-----------------------+----------------------------------------------+------------+-------------+-------------+---------------------+----------------------------------------------------------------------+----------------+-----------------+---------------------------------------+-------------------+------------------+--------------+
| id | pull_request_id | license_id | conflict_id | repository_license_id | file_path                                    | match_type | match_value | match_lines | component_file_path | component_file_url                                                   | component_name | component_lines | component_url                         | component_version | component_vendor | issue        |
+----+-----------------+------------+-------------+-----------------------+----------------------------------------------+------------+-------------+-------------+---------------------+----------------------------------------------------------------------+----------------+-----------------+---------------------------------------+-------------------+------------------+--------------+
|  8 |              11 |          6 |        NULL |                  NULL | internal/controller/scoringmgr/scoringmgr.go | snippet    | 63%         | 53-145      | fs/exfat/misc.c     | https://osskb.org/api/file_contents/0e0cc80e43dc53f42483f6aa8e295f7b | cloud-kernel   | 1-93            | https://gitee.com/anolis/cloud-kernel | 5.19.0-1_rc1      | anolis           | 0x01         |
|  9 |              11 |          4 |        NULL |                  NULL | internal/controller/scoringmgr/scoringmgr.go | snippet    | 63%         | 53-145      | fs/exfat/misc.c     | https://osskb.org/api/file_contents/0e0cc80e43dc53f42483f6aa8e295f7b | cloud-kernel   | 1-93            | https://gitee.com/anolis/cloud-kernel | 5.19.0-1_rc1      | anolis           | 0x01         |
| 10 |              11 |       NULL |           1 |                     1 | NULL                                         | NULL       | NULL        | NULL        | NULL                | NULL                                                                 | NULL           | NULL            | NULL                                  | NULL              | NULL             | 0x01         |
+----+-----------------+------------+-------------+-----------------------+----------------------------------------------+------------+-------------+-------------+---------------------+----------------------------------------------------------------------+----------------+-----------------+---------------------------------------+-------------------+------------------+--------------+

```

**Review by tiokim — COMMENTED** [review]
PTAL

**Comment by o-konoval** [comment]
I tried to check work of this PR.
I cloned this PR on my VM and follow steps for docker build and run-before filled data in application.properties.

There are PRs in DB:
mysql> select * from pull_requests;
+----+----------------------------+-------------+----------------------+-------------------------------------------------+-----------------------------------------+-----------------------------+-------------------+-------------------+------------+
| id | scan_date | user | repository_name | url | diff_url | status | pull_request_head | pull_request_base | sender |
+----+----------------------------+-------------+----------------------+-------------------------------------------------+-----------------------------------------+-----------------------------+-------------------+-------------------+------------+
| 1 | 2023-10-09 14:34:44.067000 | GitHub hook | userai-dev/dockerapp | https://github.com/userai-dev/dockerapp/pull/35 | https://github.com/userai-dev/dockerapp | Scan completed | userai-dev | userai-dev | userai-dev |
| 2 | 2023-10-09 15:20:01.602000 | GitHub hook | userai-dev/dockerapp | https://github.com/userai-dev/dockerapp/pull/36 | https://github.com/userai-dev/dockerapp | Error while posting results | userai-dev | userai-dev | userai-dev |
| 3 | 2023-10-09 15:21:30.574000 | GitHub hook | userai-dev/dockerapp | https://github.com/userai-dev/dockerapp/pull/37 | https://github.com/userai-dev/dockerapp | Scan completed | userai-dev | userai-dev | userai-dev |
+----+----------------------------+-------------+----------------------+-------------------------------------------------+-----------------------------------------+-----------------------------+-------------------+-------------------+------------+
3 rows in set (0.00 sec)

But no information on FE:in My Pull Request&My Repo PR-however- PR was handled successfully:
lpvs-lpvs-1 | 2023-10-09 15:21:30.573 INFO 1 --- [ main] c.l.service.LPVSQueueProcessorService : PROCESS Webhook id = 3
lpvs-lpvs-1 | 2023-10-09 15:21:30.575 INFO 1 --- [ LPVS-ASYNC::2] com.lpvs.service.LPVSQueueService : GitHub Webhook processing...
lpvs-lpvs-1 | 2023-10-09 15:21:30.577 INFO 1 --- [ LPVS-ASYNC::2] com.lpvs.service.LPVSQueueService : com.lpvs.entity.LPVSQueue@510155fa

I filled fields by my Google account info-may be something missed -however I successfully authorized on localhost:7896. 
I am sure that You do not have such problem- what may be wrong?

**Comment by tiokim** [comment]
> I tried to check work of this PR. I cloned this PR on my VM and follow steps for docker build and run-before filled data in application.properties.
> 
> There are PRs in DB: mysql> select * from pull_requests; +----+----------------------------+-------------+----------------------+-------------------------------------------------+-----------------------------------------+-----------------------------+-------------------+-------------------+------------+ | id | scan_date | user | repository_name | url | diff_url | status | pull_request_head | pull_request_base | sender | +----+----------------------------+-------------+----------------------+-------------------------------------------------+-----------------------------------------+-----------------------------+-------------------+-------------------+------------+ | 1 | 2023-10-09 14:34:44.067000 | GitHub hook | userai-dev/dockerapp | [userai-dev/dockerapp#35](https://github.com/userai-dev/dockerapp/pull/35) | https://github.com/userai-dev/dockerapp | Scan completed | userai-dev | userai-dev | userai-dev | | 2 | 2023-10-09 15:20:01.602000 | GitHub hook | userai-dev/dockerapp | [userai-dev/dockerapp#36](https://github.com/userai-dev/dockerapp/pull/36) | https://github.com/userai-dev/dockerapp | Error while posting results | userai-dev | userai-dev | userai-dev | | 3 | 2023-10-09 15:21:30.574000 | GitHub hook | userai-dev/dockerapp | [userai-dev/dockerapp#37](https://github.com/userai-dev/dockerapp/pull/37) | https://github.com/userai-dev/dockerapp | Scan completed | userai-dev | userai-dev | userai-dev | +----+----------------------------+-------------+----------------------+-------------------------------------------------+-----------------------------------------+-----------------------------+-------------------+-------------------+------------+ 3 rows in set (0.00 sec)
> 
> But no information on FE:in My Pull Request&My Repo PR-however- PR was handled successfully: lpvs-lpvs-1 | 2023-10-09 15:21:30.573 INFO 1 --- [ main] c.l.service.LPVSQueueProcessorService : PROCESS Webhook id = 3 lpvs-lpvs-1 | 2023-10-09 15:21:30.575 INFO 1 --- [ LPVS-ASYNC::2] com.lpvs.service.LPVSQueueService : GitHub Webhook processing... lpvs-lpvs-1 | 2023-10-09 15:21:30.577 INFO 1 --- [ LPVS-ASYNC::2] com.lpvs.service.LPVSQueueService : com.lpvs.entity.LPVSQueue@510155fa
> 
> I filled fields by my Google account info-may be something missed -however I successfully authorized on localhost:7896. I am sure that You do not have such problem- what may be wrong?

Have you set `Github ID` from the `/user/setting` page with your github account, `userai-dev`?

File: frontend/src/css/Result_style.css [hunk 3]
```diff
@@ -0,0 +1,1662 @@
+/**
+ * Copyright 2023 kyudori, Basaeng, hwan5180, quswjdgma83
+ *
+ * Use of this source code is governed by a MIT license that can be
+ * found in the LICENSE file.
+ */
+
+.result-pull-request {
```
Comment (MoonkiHong) [inline frontend/src/css/Result_style.css:8]: Please check out its indentation. (See your similar definition below)

**Review by MoonkiHong — COMMENTED** [review]
Please take a look at my comment.

**Comment by o-konoval** [comment]
> Have you set `Github ID` from the `/user/setting` page with your github account, `userai-dev`?

Yes, I obtained GitHub ID and entered it in Profile during login on localhost:7896
I re-check all again and unfortunately no changes

File: frontend/src/css/Result_style.css (hunk 3 above)
Comment (kyudori) [reply frontend/src/css/Result_style.css:8]: Modified indentation overall.

File: frontend/src/pages/Result.jsx (hunk 2 above)
Comment (kyudori) [reply frontend/src/pages/Result.jsx]: Fixed an error for NULL.

File: frontend/src/pages/History.jsx (hunk 1 above)
Comment (kyudori) [reply frontend/src/pages/History.jsx:199]: Done.

**Review by o-kopysov — APPROVED** [review]
LGTM

**Review by m-rudyk — APPROVED** [review]
I have no objection, approved.

**Comment by tiokim** [comment]
> 
> Yes, I obtained GitHub ID and entered it in Profile during login on localhost:7896. I re-check all again and unfortunately no changes.

Can you share your records of the `member` table too?

**Review by tiokim — COMMENTED** [review]
@kyudori Can you revert `Home_style.css` and `Login_style.css` files? There was no problem before, but it seems to have changed.

**Comment by kyudori** [comment]
> @kyudori Can you revert `Home_style.css` and `Login_style.css` files? There was no problem before, but it seems to have changed.

Reverting Home_style.css and Login_style.css to previous commit.

**Review by tiokim — APPROVED** [review]
LGTM

**Comment by o-konoval** [comment]
> > Yes, I obtained GitHub ID and entered it in Profile during login on localhost:7896. I re-check all again and unfortunately no changes.
> 
> Can you share your records of the `member` table too?
mysql> select * from member;

+----+------------------------------+--------+----------+----------+--------------+

| id | email                        | name   | nickname | provider | organization |

+----+------------------------------+--------+----------+----------+--------------+

|  1 | oleoleole123456789@gmail.com | Oleg K | NULL     | google   | NULL         |

+----+------------------------------+--------+----------+----------+--------------+

1 row in set (0.00 sec)

**Comment by tiokim** [comment]
> 
> +----+------------------------------+--------+----------+----------+--------------+
> 
> | id | email | name | nickname | provider | organization |
> 
> +----+------------------------------+--------+----------+----------+--------------+
> 
> | 1 | [oleoleole123456789@gmail.com](mailto:oleoleole123456789@gmail.com) | Oleg K | NULL | google | NULL |
> 
> +----+------------------------------+--------+----------+----------+--------------+
> 

As you can see, the `nickname` and the `organization` are NULL. If you put your `github id`, userai-dev, to the `nickname`, you can see the appropriate web page.

**Comment by o-konoval** [comment]
Thanks a lot. Now it works- I was confused by GitHub ID value - I read that GitHub ID may be found by request to API- it is an unique number.

GitHub ID may be found in answer of request:
https://api.github.com/users/your_github_user_name
where instead of your_github_user_name you must use the desired GitHub username. 

So if possible it would be great if instead GitHub ID on authorization page would be GitHub username or GitHub login- however may be it is my understanding of this term.

**Pull request merged**